zilliqa/src/api/types/zil.rs

```diff
@@ -396,3 +396,118 @@ pub enum RPCErrorCode {
     RpcInWarmup = -28,             // Client still warming up
     RpcMethodDeprecated = -32,     // RPC method is deprecated
 }
+
+#[derive(Serialize, Deserialize, Clone)]
+pub struct DSBlock {
+    pub header: DSBlockHeader,
+    pub signature: String,
+}
+
+impl From<DSBlockVerbose> for DSBlock {
+    fn from(verbose_block: DSBlockVerbose) -> Self {
+        DSBlock {
+            header: DSBlockHeader::from(verbose_block.header),
+            signature: verbose_block.signature,
+        }
+    }
+}
+
+#[allow(non_snake_case)]
+#[derive(Serialize, Deserialize, Clone)]
+pub struct DSBlockHeader {
+    pub BlockNum: String,
+    pub Difficulty: u64,
+    pub DifficultyDS: u64,
+    pub GasPrice: String,
+    pub PoWWinners: Vec<String>,
+    pub PrevHash: String,
+    pub Timestamp: String,
+}
+
+#[allow(non_snake_case)]
+#[derive(Serialize, Deserialize, Clone)]
+pub struct DSBlockVerbose {
+    // Sample fields based on given/expected data structure
+    pub B1: Vec<bool>,
+    pub B2: Vec<bool>,
+    pub CS1: String,
+    pub PrevDSHash: String,
+    pub header: DSBlockHeaderVerbose,
+    pub signature: String,
+}
+
+impl From<DSBlockHeaderVerbose> for DSBlockHeader {
+    fn from(header: DSBlockHeaderVerbose) -> Self {
+        DSBlockHeader {
+            BlockNum: header.BlockNum,
+            Difficulty: header.Difficulty,
+            DifficultyDS: header.DifficultyDS,
+            GasPrice: header.GasPrice,
+            PoWWinners: header.PoWWinners,
+            PrevHash: header.PrevHash,
+            Timestamp: header.Timestamp,
+        }
+    }
+}
+
+#[allow(non_snake_case)]
+#[derive(Serialize, Deserialize, Clone)]
+pub struct DSBlockHeaderVerbose {
+    pub BlockNum: String,
```

Comment:
I think the advantage of avoiding duplication in this sort of situation would be so that if you change the non-verbose API you don't forget the verbose one, but given the DS apis aren't expected to change it probably doesn't make too much difference here?